### ISSUE TITLE
dotnet publish ignores the target runtime

### ISSUE THREAD

**Issue by klmcwhirter** [aspnet/Microsoft.Data.Sqlite#383]
When doing cross-compliation to another platform and using the dotnet publish -r {RID} command the resulting SQLite shared object is not being published correctly.

Development Platform: macOS Sierra
dotnet -v: .NET Command Line Tools (2.0.0-preview2-006497)

Target Platform: Raspberry Pi 3 - Ubuntu 16.04 MATE

Command: dotnet publish --self-contained -r {RID}
_Note I have used linux-arm and ubuntu.16.04-arm as the RID without success_

Result: libe_sqlite3.dylib is emitted into the publish directory instead of libe_sqlite3.so as expected - this results in the error _System.DllNotFoundException: Unable to load DLL 'e_sqlite3'_. Note that I have worked through all of the dependencies. I had it working with preview1 by copying the .so file that @ericsink made available in the armhf dir mentioned below. With preview2, however, I cannot make it work. I have stopped here,

_Note that the appropriate lib*.so has been made available by @ericsink in directories at : [linux binaries](https://github.com/ericsink/SQLitePCL.raw/tree/master/linux) - in my case the binary in armhf is applicable.

What other information do you need?

**Comment by ericsink** [comment]
I wonder if this problem is still present if ARM is taken out of the picture? What happens if the target platform is Linux x64, or something windows?

I'll try this myself when I get a chance. For now I just wanted to express the curiosity.

**Comment by klmcwhirter** [comment]
Good line of thinking!

My standard deployment target is a docker image based on "jessie" debian on x86_64 and so I use the debian.8-x64 RID to restore, build and publish.

This is the publish command I am using:
```bash
dotnet publish --self-contained -r debian.8-x64
```

I inspected my latest image and both the dylib and .so are being published. I inspected the result of the publish operation and it produces the same result (of course, just wanted to be sure).

```bash
root@7b3bc8966f70:/app# ls -l libe_sqlite3*
-rw-r--r-- 1 root root 2454008 Jun  4 00:02 libe_sqlite3.dylib
-rw-r--r-- 1 root root  852965 Jun  4 00:02 libe_sqlite3.so
root@7b3bc8966f70:/app# 
```

**Comment by klmcwhirter** [comment]
Here is my .csproj file in case you need that:
```
<Project Sdk="Microsoft.NET.Sdk.Web">
  <PropertyGroup>
    <TargetFramework>netcoreapp2.0</TargetFramework>
    <OutputType>Exe</OutputType>
  </PropertyGroup>
  <ItemGroup>
    <None Include="appsettings.json" CopyToOutputDirectory="Always" CopyToPublishDirectory="Always" />
    <None Update="appsettings.json;Dockerfile.debug;Dockerfile;docker-compose.debug.yml;docker-compose.yml">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
      <CopyToPublishDirectory>Always</CopyToPublishDirectory>
    </None>
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Autofac" Version="4.6.0" />
    <PackageReference Include="Autofac.Extensions.DependencyInjection" Version="4.1.0" />
    <PackageReference Include="Microsoft.AspNetCore.All" Version="2.0.0-preview2-final" />
    <PackageReference Include="System.Drawing.Primitives" Version="4.3.0" />
  </ItemGroup>
  <ItemGroup>
    <DotNetCliToolReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Tools" Version="2.0.0-preview2-final" />
    <DotNetCliToolReference Include="Microsoft.EntityFrameworkCore.Tools.DotNet" Version="1.0.1" />
  </ItemGroup>
</Project>
```

_I am purposely avoiding mentioning the problems I am having after upgrading to preview2 in this issue. I will open another issue somewhere once I dig a little deeper._

**Comment by ericsink** [comment]
I realized that part of what is going on here is probably the same issue we recently fixed in SQLitePCLRaw 1.1.7.

Specifically using 1.1.3, I can reproduce behavior similar to what is described above.  On Windows, when I try to dotnet publish for Linux x64 or Mac, I also get the Windows versions of e_sqlite3.dll in the publish output.  When I update to version 1.1.7 of the SQLitePCLRaw bundle, this problem does not appear.

Adding ARM into this story results in other problems that are, er, ARM-specific.  Version 1.1.7 does not have ARM binaries for e_sqlite3.

**Comment by klmcwhirter** [comment]
Good news. I am assuming the issue with the dylib always being emitted is somewhere else? Should I open a ticket with the dotnet cli project? Or something else? Have you already done that?

Thanks again!

Will 1.1.7 be in the preview3 or release candidate?

I look forward to it. I am almost there ...

Now just need to get Docker to support automated builds for the ARM platform ...

**Comment by bricelam** [comment]
> Will 1.1.7 be in the preview3 or release candidate?

Yes. Whatever the next release ends up being called, it will be in that. You can always use the [nightly builds](https://dotnet.myget.org/gallery/aspnetcore-dev) to try it now.

**Comment by klmcwhirter** [comment]
Good to know that the Microsoft.Data.Sqlite package has been updated to use 1.1.7+. I will checkout 2.0.0-rtm-26180 today.
Thanks.

**Comment by klmcwhirter** [comment]
FYI - while I am disappointed to see official support for the ARM platform in 2.0, I do understand.

I also was able to confirm (finally) that publishing for linux-arm no longer causes the libe_sqlite3.dylib to be copied into place.
